skins/oasis/modules/PageHeaderController.class.php

```diff
@@ -638,4 +638,28 @@ static function onArticleSaveComplete( &$article, &$user, $text, $summary,
 		$wgMemc->delete( wfMemcKey( 'mOasisRecentRevisions2', $article->getTitle()->getArticleId() ) );
 		return true;
 	}
+
+	private function renderTemplateSubtitle() {
+		global $wgUser;
+		$templateType = $this->getTemplateType();
+		if ( $wgUser->isLoggedIn() ) {
+			$templateType .= $this->renderEditButton();
+			return $templateType;
+		} else {
+			if ( $templateType !== 'Unclassified' ) {
+				return $templateType;
+			}
+			return wfMessage( 'oasis-page-header-subtitle-template' )->escaped();
+		}
+
+	}
+
+	private function getTemplateType() {
```

Comment:
Can we move the method outside of the `PageHeaderController` as it's strictly connected to TC?